crates/starfish/core/src/authority_service.rs

```diff
@@ -449,11 +449,6 @@ impl<C: CoreThreadDispatcher> NetworkService for AuthorityService<C> {
             .map_err(|_| ConsensusError::Shutdown)?;
 
         // 9. Add block to dag, add its missing ancestors to the set
-        // TODO:: consider possible optimization:
-        //  first try to accept the block. If it fails, try to find missing ancestors
-        //  among additional headers and from block_round-1 add only them. From the
-        //  rounds < block_round-1 add all headers
-        // TODO: handle missing transactions as well
```

Comment:
missing transactions are handled right after step 9 and before return from method.